pkg/ingester/tailer.go

```diff
@@ -102,6 +115,37 @@ func (t *tailer) loop() {
 	}
 }
 
+func (t *tailer) receiveStreamsLoop() {
+	for {
+		select {
+		case <-t.conn.Context().Done():
+			t.close()
```

Comment:
I think maybe t.close() should be in a `defer` block so that anywhere we return from this function it would be called?

Reply:
Done